pkg/controllers/dlb/controller.go

```diff
@@ -76,6 +76,7 @@ func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	devicePlugin := rawObj.(*devicepluginv1.DlbDevicePlugin)
 
 	daemonSet := deployments.DLBPluginDaemonSet()
+	daemonSet.Spec.Template.Spec.NodeSelector = controllers.ReplaceToDefaultNodeSelectorIfEmpty(devicePlugin.Spec.NodeSelector)
```

Comment:
I'd suggest to use previous solution:
```
if len(devicePlugin.Spec.NodeSelector) > 0 {
        daemonSet.Spec.Template.Spec.NodeSelector = devicePlugin.Spec.NodeSelector
}
```
- it's much more clear, explicit and readable
- we don't need extra file api.go
- we don't need to parse whole daemonset yaml in UpdateDaemonSet just to get node selector